src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -59,75 +60,68 @@ trait RawlsBillingProjectComponent {
+            }
+          }
       }
     }
 
     def loadProjectUsersWithEmail(rawlsProjectName: RawlsBillingProjectName): ReadAction[Seq[RawlsBillingProjectMember]] = {
-      val name = rawlsProjectName.value
-
-      val query = for {
-        projectUser <- projectUsersQuery if projectUser.projectName === name
-        user <- rawlsUserQuery if user.userSubjectId === projectUser.userSubjectId
-      } yield (user.userEmail, projectUser.role)
+      val ownerUsersQuery = findProjectUsersWithRole(rawlsProjectName, ProjectRoles.Owner)
+      val userUsersQuery = findProjectUsersWithRole(rawlsProjectName, ProjectRoles.User)
 
-      query.result.map(_.map { case (email, role) =>
-        RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))
-      })
-    }
-
-    def delete(billingProject: RawlsBillingProject): ReadWriteAction[Boolean] = {
-      val name = billingProject.projectName.value
-      val projectQuery = findBillingProjectByName(name)
+      val ownerSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.Owner)
+      val userSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.User)
 
-      uniqueResult[RawlsBillingProjectRecord](projectQuery).flatMap {
-        case None => DBIO.successful(false)
-        case Some(projectRec) =>
-          findUsersByProjectName(name).delete andThen projectQuery.delete map { count => count > 0 }
-      }
-    }
+      val owners = ownerUsersQuery union ownerSubgroupsQuery
+      val users = userUsersQuery union userSubgroupsQuery
 
-    def addUserToProject(userRef: RawlsUserRef, billingProjectName: RawlsBillingProjectName, role: ProjectRole): WriteAction[ProjectUsersRecord] = {
-      val record = marshalProjectUsers(userRef, billingProjectName, role)
-      projectUsersQuery insertOrUpdate record map { _ => record }
+      (owners union users).result.map(_.map{ case(email, role) => RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))})
     }
 
-    def removeUserFromProject(userRef: RawlsUserRef, billingProject: RawlsBillingProject): WriteAction[Boolean] = {
-      val query = projectUsersQuery.filter(q => q.userSubjectId === userRef.userSubjectId.value && q.projectName === billingProject.projectName.value)
-      query.delete.map { count => count > 0 }
+    def delete(billingProjectName: RawlsBillingProjectName): ReadWriteAction[Boolean] = {
+      findBillingGroups(billingProjectName).result.flatMap { groupNames =>
+        rawlsBillingProjectGroupQuery.filter(_.projectName === billingProjectName.value).delete andThen
+          DBIO.sequence(groupNames.map { groupName => rawlsGroupQuery.delete(RawlsGroupRef(RawlsGroupName(groupName)))}) andThen
+            rawlsBillingProjectQuery.filter(_.projectName === billingProjectName.value).delete map { count => count > 0 }
+      }
     }
 
-    def removeUserFromAllProjects(userRef: RawlsUserRef): WriteAction[Boolean] = {
```

Comment:
this was only used for deleteUser. deleteUser already removes a user from all groups, so this bit ends up being handled